File: sharedemo/src/main/java/com/example/administrator/sharedemo/MainActivity.java
package com.example.administrator.sharedemo;

import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {
    MyView myCustomView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        myCustomView = findViewById(R.id.custom);
        Log.i("chuan", "onCreate init myCustomView  width=" + myCustomView.getWidth());
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("chuan", "Main onResume");
        new Handler().post(new Runnable() {
            @Override
            public void run() {
                Log.i("chuan", "onResume Handler post runnable button width=" + myCustomView.getWidth());
            }
        });
        myCustomView.post(new Runnable() {
            @Override
            public void run() {
                Log.i("chuan", "onResume myCustomView post runnable button width=" + myCustomView.getWidth());
            }
        });
    }
}

File: authdemo/src/main/java/com/youdianpinwei/authdemo/MainActivity.kt
package com.youdianpinwei.authdemo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import android.widget.Button


class MainActivity : AppCompatActivity() {
    private var mBt: Button? = null
    private var mUser_package_name: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mBt = findViewById(R.id.bt_confirm_login)
        mBt?.setOnClickListener({
            val intent1 = Intent()
            intent1.setAction("kx.com.kx.bapp.sign")
            intent1.putExtra("token", "xxxx-xxxx-xxxxx")
            intent1.putExtra("app_pkg", "com.kx.aapp")
            sendBroadcast(intent1)
            finish()
        })
        mBt?.isEnabled = false

        //获取传递的数据
        val intent = intent
        val uri = intent.data

        //获取参数值
        val type = uri.getQueryParameter("type")
        mUser_package_name = uri.getQueryParameter("user_package_name")

        //类型type 检验
        if (TextUtils.equals(type, "1")) {
            //Todo 未登录 处理
            request5037()
        }
    }

    /**
     * 外部 app 拉起授权 (code:5037)
     * 服务器校验 是否授权
     * 授权 页面A状态
     */

    private fun request5037() {
        mBt?.setEnabled(true)
    }

}
